File: infra/components/vertex_ai.py
"""Vertex AI TensorBoard for experiment tracking.

Note: AiMetadataStore intentionally omitted — google-beta provider bug causes
a perpetual create/destroy cycle. See DEPLOY.md for history.
"""

import pulumi
import pulumi_gcp as gcp


class VertexAI(pulumi.ComponentResource):
    """Vertex AI resources for ML experiment tracking and visualization."""

    tensorboard_id: pulumi.Output[str]

    def __init__(
        self,
        name: str,
        project_id: str,
        region: str,
        opts: pulumi.ResourceOptions | None = None,
    ):
        super().__init__("genomics:infra:VertexAI", name, None, opts)

        child = pulumi.ResourceOptions(parent=self)

        self.tensorboard = gcp.vertex.AiTensorboard(
            f"{name}-tensorboard",
            project=project_id,
            region=region,
            display_name="precision-genomics-tensorboard",
            opts=child,
        )

        self.tensorboard_id = self.tensorboard.name

        self.register_outputs(
            {
                "tensorboard_id": self.tensorboard_id,
            }
        )

File: infra-ts/components/vertexAi.ts
// Note: AiMetadataStore intentionally omitted — google-beta provider bug
// causes a perpetual create/destroy cycle. See DEPLOY.md for history.

import * as pulumi from "@pulumi/pulumi";
import * as gcp from "@pulumi/gcp";

export class VertexAI extends pulumi.ComponentResource {
  public readonly tensorboardName: pulumi.Output<string>;

  constructor(
    name: string,
    args: { projectId: string; region: string },
    opts?: pulumi.ComponentResourceOptions,
  ) {
    super("genomics:infra:VertexAI", name, {}, opts);
    const child = { parent: this };

    const tensorboard = new gcp.vertex.AiTensorboard(`${name}-tensorboard`, {
      displayName: "precision-genomics-tensorboard",
      project: args.projectId,
      region: args.region,
    }, child);

    this.tensorboardName = tensorboard.name;

    this.registerOutputs({
      tensorboardName: this.tensorboardName,
    });
  }
}
